### PR TITLE
Load any validation errors surfaced by the pipeline and display them on the sample page

### PR DESCRIPTION
**Pull request by cdebourcy**
When available, this can provide visibility into the reason for the failure and save on-call work.

image

Summoning @happyimadesignr to run the frontend addition by her (see screenshot).

### CODE REVIEW

File: app/models/pipeline_run.rb [hunk 1]
```diff
@@ -393,6 +395,15 @@ def succeeded?
     job_status == STATUS_CHECKED
   end
 
+  def db_load_input_validations
+    file = Tempfile.new
+    downloaded = PipelineRun.download_file_with_retries(s3_file_for("input_validations"),
+                                                        file.path, 3, false)
+    error_message = downloaded ? JSON.parse(File.read(file))["Validation error"] : nil
+    update(error_message: error_message) if error_message
```
Comment (jshoe) [inline app/models/pipeline_run.rb]: may want to append to existing error_message if we use this field for other messages in the future

File: app/models/pipeline_run.rb [hunk 2]
```diff
@@ -967,22 +980,25 @@ def local_amr_drug_summary_path
...
     end
   end
 
-  def self.download_file(s3_path, destination_dir)
-    command = "mkdir -p #{destination_dir};"
-    command += "aws s3 cp #{s3_path} #{destination_dir}/;"
-    _stdout, _stderr, status = Open3.capture3(command)
-    return nil unless status.exitstatus.zero?
-    "#{destination_dir}/#{File.basename(s3_path)}"
+  def self.download_file(s3_path, destination, dest_is_dir = true)
```
Comment (MarkAZhang) [inline app/models/pipeline_run.rb]: Maybe add a comment "If the destination path is a directory, create the directory if necessary and append the file name to the end of the destination path. If the destination path includes a file name, assume that the parent directories already exist."

Comment (MarkAZhang) [reply app/models/pipeline_run.rb]: Took a little while to understand this function.

File: app/assets/src/components/views/SampleView/SampleView.jsx [hunk 3]
```diff
@@ -296,6 +296,9 @@ class SampleView extends React.Component {
         return (
           <div className={cs.failedContainer}>
             <h6 className={cs.failed}>Sample run failed</h6>
+            <h6 className={cs.failed}>
+              {this.props.pipelineRun.error_message}
```
Comment (jshoe) [inline app/assets/src/components/views/SampleView/SampleView.jsx]: Design-wise the error message could have a separate style to be more friendly (just normalcase, non-red)

File: app/models/pipeline_run.rb (hunk 2 above)
Comment (cdebourcy) [reply app/models/pipeline_run.rb]: ok, added your comment

File: app/models/pipeline_run.rb (hunk 1 above)
Comment (cdebourcy) [reply app/models/pipeline_run.rb]: good point

File: app/assets/src/components/views/SampleView/SampleView.jsx (hunk 3 above)
Comment (cdebourcy) [reply app/assets/src/components/views/SampleView/SampleView.jsx]: true

**Comment by happyimadesignr** [comment]
Thanks for including me @cdebourcy. We have this style for errors that @MarkAZhang created.

At some point, it would be nice to take some time to audit all the places we have error states and make them consistent.

Screen Shot 2019-04-23 at 9 48 45 AM

**Pull request merged**